### DISCUSSION TITLE
简化 Ant Engine 构建新项目的流程

### DISCUSSION THREAD

**Discussion by cloudwu** [ejoy/ant#124] [General]
最近几天，我们简化了 Ant Engine 构建新项目的流程。

目前，如果在 Windows 下开发，游戏本身的 C/C++ 代码不需要和 Ant Engine 的源代码一起编译链接了。游戏的 C/C++ 代码以动态库 DLL 的形式存在，只需要链接 lua 的动态库。（可以使用开发者自己喜爱的任意构建工具）

所以，定期发布 Ant Engine 的已编译好版本可以供开发者直接使用，而从源代码构建项目变成一个可选项。

@constly 建议你更新一下你的 demo 项目 https://github.com/constly/ant_demo  :)

**Comment by constly** [comment]
好的，我晚上更新看看，我最近做的一些尝试都没往里面同步了，正好一起同步一波

**Reply by cloudwu** [reply]
文档还没有补充。看看有什么问题直接这里提。理论上比之前要简单很多。

**Reply by constly** [reply]
我在仿照vaststars里面的写法，我目前卡在这里: [make.lua](https://github.com/constly/ant_demo/blob/main/startup/com/clibs/ly.imgui.extend/make.lua)，我项目里面扩展了imgui，现在会报一堆无法解析的外部符号。 我继续研究看看

**Reply by constly** [reply]
我改为之前的构建流程：将我的代码与引擎源代码一起编译为新的exe，这次倒是正常的。

**Reply by constly** [reply]
有个问题，之前我看windows下区分了编辑器版本和运行时版本，最新版是没有了吗，make.lua里面没看到ant_editor了~

**Reply by cloudwu** [reply]
现在两个版本已经合并成同一个 exe 了。

**Reply by constly** [reply]
好的，那合并为一个版本后，如何修改启动脚本，支持多开呢~

**Reply by actboy168** [reply]
`ant.exe -rt`就等价于原来的`ant.exe`

**Comment by constly** [comment]
另外一个问题，mount文件的规则可否优化下，比如我在.mount中新加了一个game目录
```
mount:
    /engine/ %engine%/engine
    /pkg/    %engine%/pkg
    /        %project%
    /        %project%/mod
    /        %project%/game
```
我必须得在game目录下再建一个名为pkg的子目录，然后把pkg包放里面，感觉这个规则太死了，可以允许我直接把包放game目录下~

**Reply by cloudwu** [reply]
@actboy168 斟酌一下。

不过我的看法是，如果直接放在 game 目录下，或许会和非 pkg 的内容产生混淆？

**Reply by actboy168** [reply]
可以这样写
``` yaml
mount:
    /engine/ %engine%/engine
    /pkg/    %engine%/pkg
    /        %project%
    /        %project%/mod
    /pkg/    %project%/game
```

**Reply by constly** [reply]
明白了，我的锅，我之前尝试时不仔细，缺了package.ecs，没把流程跑通。刚才又试了下可以了

**Comment by cloudwu** [comment]
> 我在仿照vaststars里面的写法，我目前卡在这里: [make.lua](https://github.com/constly/ant_demo/blob/main/startup/com/clibs/ly.imgui.extend/make.lua)，我项目里面扩展了imgui，现在会报一堆无法解析的外部符号。 我继续研究看看

这个需求比较特殊，因为我假设游戏并不会直接使用 imgui 的 c api 。如果硬要做，需要学 lua 那样把 api 导出：Lua 目前 build 出了一个桥接的 dll ，把符号从 exe 嫁接到一个独立 dll 上。游戏 link 这个 dll 就可以了。

**Reply by constly** [reply]
了解，我倒不纠结怎么编译，目前我还是与引擎源码一起编译得了~
